docs/source/sphinxext/pre_sphinx_gallery.py

```diff
@@ -0,0 +1,177 @@
+"""
+Override sphinx_gallery's treatment of groups (folders) with cartopy's
+``__tags__`` semantics. This is tightly bound to the sphinx_gallery
+implementation, hence the explicit version checking.
+
+"""
+from collections import OrderedDict
+import os.path
+import shutil
+import tempfile
+import textwrap
+
+import sphinx_gallery.gen_gallery
+import sphinx_gallery.gen_rst
+from sphinx_gallery.gen_rst import (
+    write_backreferences, extract_intro, _thumbnail_div,
+    generate_file_rst, sphinx_compatibility)
+
+
+if sphinx_gallery.__version__ not in ['0.1.12']:
+    raise RuntimeError('not tested with this version of sphinx_gallery ({}). '
+                       'Please modify this check, and validate sphinx_gallery'
+                       ' behaves as expected.'
+                       ''.format(sphinx_gallery.__version__))
+
+
+GALLERY_HEADER = textwrap.dedent("""
+
+    Cartopy Gallery
+    ---------------
+
+    The following visual examples demonstrate some of the functionality of
+    Cartopy, particularly its matplotlib interface.
+
+    For a structured introduction to cartopy, including some of these
+    examples, see :ref:`getting-started-with-cartopy`.
+
+""")
+
+
+def example_groups(src_dir):
+    """Return a dictionary of {tag: [example filenames]} for the given dir."""
+
+    sorted_listdir = [fname for fname in sorted(os.listdir(src_dir))
+                      if fname.endswith('.py') and not fname.startswith('_')]
+    tagged_examples = {}
+
+    for fname in sorted_listdir:
+        fpath = os.path.join(src_dir, fname)
+        __tags__ = []
```

Comment:
I added this line in to appease flake8, but ended up breaking the functionality (``__tags__`` always remains empty).
I only saw this when I did a clean build again locally. 😢 

Will fix in #1027.